### ISSUE TITLE
cannot set num-search-threads

### ISSUE THREAD

**Issue by kylebakerio** [sanderland/katrain#610]
tldr:

num-analysis-threads is a command line option. It appears to be set in config.json--I would guess that's what "threads" corresponds to. Its default value for me is 12, and that is what katago reports on initialization.

Unfortunately, modifying the threads value doesn't seem to do anything; everytime I start up katrain, it overwrites that value back to 12 silently.

????

-----

More details you probably don't need:

I'm using katago and katrain both as cloned down from github; I'm on pop os 22.04, an ubuntu flavor. had trouble with libzip.so.5 when upgrading with pip from 1.10 to 1.12. Anyways, this also gets me to 1.12.4 instead of 1.12.3, so yay?

Anyways:

I'm still using the config file from ~/.katrain/config.json, and can confirm within settings and within the startup logs from katrain that this is being used. In there, at the top of the file, there's `engine.threads`, which is 12 (default value?).

I'm trying to improve hardware utilization on my machine during analysis. I use katrain/katago to review my online games. 

To fill get my vram utilization up to 80% I have to set numSearchThreads to 512 and nnMaxBatchSize to 6144. The notes recommend no higher than 256 for numSearchThreads, but without increasing `num-analysis-threads`, which seems frozen at 12, I can't get katago to use any more of my vram. (Perhaps I'm barking up the wrong tree? I'm certainly just pulling levers in the dark here, so open to hearing that.)

**Comment by sanderland** [comment]
If you changed any setting, it modifies the config it has in memory, and then saves it. That's probably what happened.

**Comment by kylebakerio** [comment]
Indeed, this does indeed seem to be the behavior. can't modify settings while katrain is open, then close katrain, then open. need to close katrain, then modify settings, then open.
